element_array_ephys/export/nwb/nwb.py

```diff
@@ -0,0 +1,637 @@
+
+        lfp.add_electrical_series(
+            pynwb.ecephys.ElectricalSeries(
+                name=f"ElectricalSeries{ephys_recording_record['insertion_number']}",
+                description=f"LFP from probe {probe_id}",
+                data=SpikeInterfaceRecordingDataChunkIterator(extractor),
+                rate=extractor.get_sampling_frequency(),
+                starting_time=(
+                    ephys_recording_record["recording_datetime"]
+                    - ephys_recording_record["session_datetime"]
+                ).total_seconds(),
+                electrodes=nwbfile.create_electrode_table_region(
+                    region=[mapping[(probe_id, x)] for x in recording_channels_by_id],
+                    name="electrodes",
+                    description="recorded electrodes",
+                ),
+                **conversion_kwargs
+            )
+        )
+
+
+def ecephys_session_to_nwb(
+    session_key,
+    raw=True,
+    spikes=True,
+    lfp="source",
+    end_frame=None,
+    lab_key=None,
+    project_key=None,
+    protocol_key=None,
+    nwbfile_kwargs=None,
+):
+    """
+    Main function for converting ephys data to NWB
+
+    Parameters
+    ----------
+    session_key: dict
+    raw: bool
+        Whether to include the raw data from source. SpikeGLX and OpenEphys are supported
+    spikes: bool
+        Whether to include CuratedClustering
+    lfp:
+        "dj" - read LFP data from ephys.LFP
+        "source" - read LFP data from source (SpikeGLX supported)
+        False - do not convert LFP
+    end_frame: int, optional
+        Used to create small test conversions where large datasets are truncated.
+    lab_key, project_key, and protocol_key: dictionaries used to look up optional additional metadata
+    nwbfile_kwargs: dict, optional
+        - If element-session is not being used, this argument is required and must be a dictionary containing
+          'session_description' (str), 'identifier' (str), and 'session_start_time' (datetime),
+          the minimal data for instantiating an NWBFile object.
+
+        - If element-session is being used, this argument can optionally be used to add over overwrite NWBFile fields.
+    """
+
+    session_to_nwb = getattr(ephys._linking_module, 'session_to_nwb', False)
+
+    if session_to_nwb:
+        nwbfile = session_to_nwb(
+            session_key,
+            lab_key=lab_key,
+            project_key=project_key,
+            protocol_key=protocol_key,
+            additional_nwbfile_kwargs=nwbfile_kwargs,
+        )
+    else:
+        nwbfile = pynwb.NWBFile(**nwbfile_kwargs)
+
+    ephys_root_data_dir = ephys.get_ephys_root_data_dir()
+
+    if raw:
+        add_ephys_recording_to_nwb(session_key, ephys_root_data_dir=ephys_root_data_dir,
+                                   nwbfile=nwbfile, end_frame=end_frame)
+
+    if spikes:
+        add_ephys_units_to_nwb(session_key, nwbfile)
```

Comment:
```suggestion
        add_ephys_units_to_nwb(session_key, nwbfile, primary_clustering_paramset_idx=primary_clustering_paramset_idx)
```

Reply:
Would suggest passing this in as an argument for the user to define.  Should change to `curation_id` pending comments above.